akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1340,6 +1341,38 @@ trait FlowOps[+Out, +Mat] {
         FlowShape(bcast.in, bcast.out(0))
     })(matF)
 
+  // TODO find this buddy a home (sounds useful enough to expose?)
+
+  def exposeAttribute[A <: Attribute]()(implicit t: ClassTag[A]): Repr[(Out, Option[A]), Mat] =
+    exposeAttribute(ConstantFun.scalaTuplify[Out, Option[A]])(t)
+
+  def exposeAttribute[A <: Attribute, E](exposeAs: (Out, Option[A]) ⇒ E)(implicit t: ClassTag[A]): Repr[E, Mat] =
+    this.via(Flow.fromGraph(new ExposeAttributeStage[Out, A, E, Mat](t.runtimeClass.asInstanceOf[Class[A]], exposeAs)))
+
+  class ExposeAttributeStage[In, A <: Attribute, E, M](attrClazz: Class[A], exposeFn: (In, Option[A]) ⇒ E)
```

Comment:
> @ktoso Isn't the affected combinator only a map? This should be quite easily be convertible into a PushStage and also combine the subsequent collect functionality into it as well.

Yeah, I'm trying exactly that right now (I'm making it a GraphStage, since the least overhead).